src/main/java/htsjdk/samtools/CRAMContainerStreamWriter.java

```diff
@@ -437,6 +457,12 @@ protected void flushContainer() throws IllegalArgumentException, IllegalAccessEx
                     final SAMRecord restoredSamRecord = f.create(cramRecords.get(i));
                     assert (restoredSamRecord.getAlignmentStart() == samRecords.get(i).getAlignmentStart());
                     assert (restoredSamRecord.getReferenceName().equals(samRecords.get(i).getReferenceName()));
+                    if (!restoredSamRecord.getReadString().equals(samRecords.get(i).getReadString())) {
```

Comment:
What is this `if` statement for? you're dumping a stacktrace, but continuing to run...what should actually happen here?